tests/bud.bats

```diff
@@ -482,37 +482,49 @@ load helpers
   test ! -s $root/vol/anothervolfile
 }
 
-@test "bud-http-Dockerfile" {
-  starthttpd ${TESTSDIR}/bud/from-scratch
+# Helper function for several of the tests which pull from http.
+#
+#  Usage:  _test_http  SUBDIRECTORY  URL_PATH  [EXTRA ARGS]
+#
+#     SUBDIRECTORY   is a subdirectory path under the 'buds' subdirectory.
+#                    This will be the argument to starthttpd(), i.e. where
+#                    the httpd will serve files.
+#
+#     URL_PATH       is the path requested by buildah from the http server,
+#                    probably 'Dockerfile' or 'context.tar'
+#
+#     [EXTRA ARGS]   if present, will be passed to buildah on the 'bud'
+#                    command line; it is intended for '-f subdir/Dockerfile'.
+#
+function _test_http() {
+  local testdir=$1; shift;        # in: subdirectory under bud/
+  local urlpath=$1; shift;        # in: path to request from localhost
+
+  starthttpd "${TESTSDIR}/bud/$testdir"
   target=scratch-image
-  run_buildah bud --signature-policy ${TESTSDIR}/policy.json -t ${target} http://0.0.0.0:${HTTP_SERVER_PORT}/Dockerfile
+  run_buildah bud --signature-policy ${TESTSDIR}/policy.json \
+              -t ${target} \
+              "$@"         \
+              http://0.0.0.0:${HTTP_SERVER_PORT}/$urlpath
   stophttpd
   run_buildah from ${target}
 }
 
+@test "bud-http-Dockerfile" {
+  _test_http from-scratch Dockerfile
+}
+
 @test "bud-http-context-with-Dockerfile" {
-  starthttpd ${TESTSDIR}/bud/http-context
-  target=scratch-image
-  run_buildah bud --signature-policy ${TESTSDIR}/policy.json -t ${target} http://0.0.0.0:${HTTP_SERVER_PORT}/context.tar
-  stophttpd
-  run_buildah from ${target}
+  _test_http http-context context.tar
 }
 
 @test "bud-http-context-dir-with-Dockerfile-pre" {
-  starthttpd ${TESTSDIR}/bud/http-context-subdir
-  target=scratch-image
-  run_buildah bud --signature-policy ${TESTSDIR}/policy.json -t ${target} -f context/Dockerfile http://0.0.0.0:${HTTP_SERVER_PORT}/context.tar
-  stophttpd
-  run_buildah from ${target}
+  _test_http http-context-subdir context.tar -f context/Dockerfile
 }
 
 @test "bud-http-context-dir-with-Dockerfile-post" {
-  # FIXME FIXME FIXME: this is 100% identical to the -pre test above.
-  starthttpd ${TESTSDIR}/bud/http-context-subdir
-  target=scratch-image
-  run_buildah bud  --signature-policy ${TESTSDIR}/policy.json -t ${target} -f context/Dockerfile http://0.0.0.0:${HTTP_SERVER_PORT}/context.tar
-  stophttpd
-  run_buildah from ${target}
+  # FIXME: this is 100% identical to the -pre test above. Why??
+  _test_http http-context-subdir context.tar -f context/Dockerfile
```

Comment:
@edsantiago I'm not sure why these are duplicates.  My only guess is they had a variance at one point that was inadvertently removed?

Reply:
I always get so embarrassed when someone points out the obvious, and said "obvious" never occurred to me. That's it.
In the original commit (7615970) the code differed as follows:
```diff
- buildah bud [options] http://etc
+ buildah bud http://etc [options]
```
(hence the `pre` and `post`)

@rhatdan changed this in #493 with the checkin comment "...not handling command options properly" and something about urfave. That made both `pre` and `post` tests identical.

I could spin up a VM and see if the `post` syntax works, or if I can find some way to make it work. Before I do so: @nalind, do you remember the reasoning behind having both `pre` and `post` testing here? Is it worth trying to recreate it?

Thank you @TomSweeneyRedHat for your insight.

Reply:
I think at one time we supported options after the image name  as well as before, perhaps with urfave.

buildah bud -t dan http://etc
buildah bud http://etc -t dan

This is no longer allowed, I believe

 buildah bud /tmp -t dan
No options (-t) can be specified after the image or container name

So I think we can remove it.

Reply:
Thank you. Dup removed, and commit message updated.